private-templates-service/adaptivecards-templating-service/src/TemplateServiceClient.ts

```diff
@@ -1174,6 +1262,32 @@ export class TemplateServiceClient {
       let tags: string[] | string = req.body.tags;
       let data: JSON[] | JSON = req.body.data;
 
+      let bindData: boolean | undefined = req.body.bindData ? req.body.bindData.toLowerCase() === "true" : undefined;
+
+      if (bindData) {
+        if (req.body.data !== undefined && (!(req.body.data instanceof Object) || !isValidJSONString(JSON.stringify(req.body.data)))) {
+          const err = new TemplateError(ApiError.InvalidTemplate, `Data must be valid JSON.`);
+          return res.status(400).json({ error: err });
+        }
+        let isClient: boolean | undefined = req.query.isClient ? req.query.isClient.toLowerCase() === "true" : undefined;
+        let token = parseToken(req.headers.authorization!);
+
+        let response = await this.bindData(
+          token,
+          req.params.id,
+          req.body.data,
+          req.body.version,
+          isClient
+        )
+
+        if (!response.success || (response.result && response.result.length === 0)) {
+          const err = new TemplateError(ApiError.DataBindingFailed, "Data binding failed.");
+          return res.status(404).json({ error: err });
```

Comment:
I wasn't sure about which error codes to use?

Reply:
Can return 400 bad request